nav2_bringup/launch/nav2_navigation_launch.py

```diff
@@ -14,91 +14,95 @@
 
 import os
 
-from launch import LaunchDescription
-import launch.actions
-
-import launch_ros.actions
-
 from ament_index_python.packages import get_package_prefix
 
 from nav2_common.launch import RewrittenYaml
 
+from launch import LaunchDescription
+from launch.conditions import IfCondition
+from launch.substitutions import LaunchConfiguration, ThisLaunchFileDir
+from launch.actions import DeclareLaunchArgument, SetEnvironmentVariable
+from launch_ros.actions import Node
+
 
 def generate_launch_description():
-    use_sim_time = launch.substitutions.LaunchConfiguration('use_sim_time', 
-                                                            default='false')
-    autostart = launch.substitutions.LaunchConfiguration('autostart')
-    params_file = launch.substitutions.LaunchConfiguration('params')
-    bt_xml_file = launch.substitutions.LaunchConfiguration('bt_xml_file')
+    use_sim_time = LaunchConfiguration('use_sim_time')
+    autostart = LaunchConfiguration('autostart')
+    params_file = LaunchConfiguration('params_file')
+    bt_xml_file = LaunchConfiguration('bt_xml_file')
+    use_lifecycle_mgr = LaunchConfiguration('use_lifecycle_mgr')
 
     # Create our own temporary YAML files that include substitutions
     param_substitutions = {
         'use_sim_time': use_sim_time,
         'bt_xml_filename': bt_xml_file,
         'autostart': autostart,
-        'map_subscribe_transient_local': 'False'
-    }
+        'map_subscribe_transient_local': 'False'}
```

Comment:
This 'map_subscribe_transient_local' needs to be overridden to 'True' when run with map_server, meaning nav2_bringup_launch.py should override it.

The 'False' default is used when running this launch file stand-alone with cartographer, since cartographer doesn't publish with transient_local.

Reply:
oh, i"ve changed this on both files.